RING_BUF_DECLARE broken for C++

The `RING_BUF_DECLARE` macro (https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/ring_buffer.h#L76-L83) initializes the `ring_buf` (defined [here](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/sys/ring_buffer.h#L34-L43)) by listing `size`, then `buffer`. This order was changed in #42845 [(here)](https://github.com/zephyrproject-rtos/zephyr/pull/42845/files#diff-caddaf4f58cb580dc17aa673b2347995552e02b5a438d9df68c9dc00c0684ca9L34-L58). C++ requires the order to match, resulting in compilation failure when using `RING_BUF_DECLARE`, which was possible before.

```
[...] zephyr/include/zephyr/sys/ring_buffer.h:83:2: error: designator order for field 'ring_buf::buffer' does not match declaration order in 'ring_buf'
```

**Environment**
 - OS: Linux
 - Current master branch of zephyr, SDK 0.13.1
 - ARM GCC
 - CONFIG_STD_CPP17=y

--- a/include/zephyr/sys/ring_buffer.h
+++ b/include/zephyr/sys/ring_buffer.h
@@ -78,8 +78,8 @@ static inline void ring_buf_internal_reset(struct ring_buf *buf, int32_t value)
 		RING_BUFFER_SIZE_ASSERT_MSG); \
 	static uint8_t __noinit _ring_buffer_data_##name[size8]; \
 	struct ring_buf name = { \
-		.size = size8, \
-		.buffer = _ring_buffer_data_##name \
+		.buffer = _ring_buffer_data_##name, \
+		.size = size8 \
 	}
 
 /**

